mesh/meshdb.go

```diff
@@ -361,41 +366,39 @@ func (m *MeshDB) writeTransactions(blk *types.Block) ([]types.TransactionId, err
 	return txids, nil
 }
 
-func (m *MeshDB) getTransactions(transactions []types.TransactionId) ([]*types.SerializableTransaction, error) {
-	var ts []*types.SerializableTransaction
-	for _, id := range transactions {
-		tBytes, err := m.getTransactionBytes(id)
-
-		if err != nil {
-			m.Error("error retrieving transaction from database ", err)
-		}
-
-		t, err := types.BytesAsTransaction(tBytes)
+func (m *MeshDB) GetTransactions(transactions []types.TransactionId) (
+	ts map[types.TransactionId]*types.SerializableTransaction,
+	mIds []types.TransactionId) {
 
+	ts = make(map[types.TransactionId]*types.SerializableTransaction, len(transactions))
+	for _, id := range transactions {
+		t, err := m.GetTransaction(id)
 		if err != nil {
-			m.Error("error deserializing transaction")
+			m.Debug("could not fetch tx %v %v", hex.EncodeToString(id[:]), err)
+			mIds = append(mIds, id)
+		} else {
+			ts[id] = t
 		}
-		ts = append(ts, t)
 	}
-
-	return ts, nil
+	return ts, mIds
 }
 
-//todo standardized transaction id across project
-//todo replace panic
-func getTransactionId(t *types.SerializableTransaction) types.TransactionId {
-	tx, err := types.TransactionAsBytes(t)
+func (m *MeshDB) GetTransaction(id types.TransactionId) (*types.SerializableTransaction, error) {
+	tBytes, err := m.getTransactionBytes(id)
 	if err != nil {
-		panic("could not Serialize transaction")
+		return nil, err
 	}
-
-	return crypto.Sha256(tx)
+	t, err := types.BytesAsTransaction(tBytes)
+	if err != nil {
+		return nil, err
+	}
+	return t, nil
 }
 
-func (m *MeshDB) getTransactionBytes(id []byte) ([]byte, error) {
-	b, err := m.transactions.Get(id)
+func (m *MeshDB) getTransactionBytes(id types.TransactionId) ([]byte, error) {
```

Comment:
I don't see that `getTransactionBytes` is used elsewhere than `GetTransaction` . if that's the case it might be better to include below code in the `GetTransaction` func. (as it's still short and you can merge the `if err` clauses )